### DISCUSSION TITLE
JUnit 5 remote test execution

### DISCUSSION THREAD

**Discussion by roded** [junit-team/junit-framework#3337] [Q&A]
Hello,
I'd like to be able to run unit tests locally but have the tests execute on a remote machine. My use case involves tests which must run on a different machine architecture and OS (i.e., these tests will fail if run locally).

Is this possible to implement using JUnit 5? I'm imagining some sort of JUnit agent running on the remote machine, accepting and running tests remotely.

Any pointers would be much appreciated.
Thanks

**Comment by marcphilipp** [comment]
JUnit does not support that natively. I'm working on a commercial product that allows distributing tests to remote machines, though: https://docs.gradle.com/enterprise/test-distribution/
